### DISCUSSION TITLE
Minimum cycle delay

### DISCUSSION THREAD

**Discussion by qulixus** [arendst/Tasmota#14828] [Q&A] [answered]
Hi all,

Just started out and pretty new to all this, so please be kind.  I had a search and couldn't find any info, or for what I needed. 

I'm using my Sonoff basics to control underfloor heating actuators via MQTT and Home Assistant. These actuators are auto balancing ones which if switched on and then off again within 60 seconds go into a reconfiguration mode, which I don't want to happen.  So what I am trying to do is set a minimum delay between power toggles, eg Turn on >> count 120 seconds before any off command is acted upon.

I know home assistant has a min_cycle_duration option but having some issues with it not working when you manually change the temperature. So would like to do it at smart plug level.

Would this be possible.  Thanks for any help/guidance

**Comment by barbudor** [comment, marked as answer]
So if I understand well, you want that once turned on, the output remains on for at least 120 seconds, ignoring any order to be turned off.
I assume that if an order to turn off occurred during the 120 seconds, you want to actually turn off at the end of the 120 seconds protection period.

My 1st idea would be to have a "virtual relay" by setting "Relay_1" to a GPIO that is not used and connected to nothing. I think this is the case for GPIO14
And set Relay_2 to the actual relay.
Assume that HA or anyone will only give orders to Relay_1 (POWER or POWER1 commands) and that Relay_2 (POWER2) should be left untouched by MQTT or Web UI

Rules such as below should do the job
```
Rule1
  ON Power1#state==1 DO backlog var1 1; Power2 1; RuleTimer1 120  ENDON
  ON Power1#state==0 DO var1 0 ENDON
  ON Rules#Timer==1 DO Power2 %var1% ENDON
Rule1 1
```
This should insure a minimum of 120 secondes ON time
Any Power1 ON command would relaunch the timer for 120 additional seconds

**Reply by qulixus** [reply]
Brilliant, thank you.  I will give that a go and report back.
